PROTO_tests/tests/join_test.py

```diff
@@ -0,0 +1,209 @@
+import numpy as np
+import pytest
+
+import arkouda as ak
+
+"""
+Encapsulates a variety of arkouda join_on_eq_with_dt test cases.
+"""
+
+
+class TestJoin:
+    @classmethod
+    def setup_class(cls):
+        cls.size = 1000
+        cls.a1 = ak.ones(cls.size, dtype=np.int64)
+        cls.a2 = ak.arange(0, cls.size, 1)
+        cls.t1 = cls.a1
+        cls.t2 = cls.a1 * 10
+        cls.dt = 10
+        ak.verbose = False
+
+    @pytest.mark.parametrize("dt_type", ["true_dt", "abs_dt", "pos_dt"])
+    def test_join_on_eq_by_dt(self, dt_type):
+        x, y = ak.join_on_eq_with_dt(self.a2, self.a1, self.t1, self.t2, self.dt, dt_type)
+        assert self.size // pytest.nl == x.size
+        assert self.size // pytest.nl == y.size
+
+    def test_join_on_eq_with_true_dt_with_result_limit(self):
+        lim = (self.size + pytest.nl) * self.size
+        res_size = self.size * self.size
+        x, y = ak.join_on_eq_with_dt(
+            self.a1, self.a1, self.a1, self.a1, self.dt, "true_dt", result_limit=lim
+        )
+        assert res_size == x.size == y.size
+
+    def test_join_on_eq_with_abs_dt_outside_window(self):
+        """
+        Should get 0 answers because N^2 matches but 0 within dt window
+        """
+        dt = 8
+        x, y = ak.join_on_eq_with_dt(self.a1, self.a1, self.t1, self.t1 * 10, dt, "abs_dt")
+        assert 0 == x.size == y.size
+
+        x, y = ak.join_on_eq_with_dt(self.a2, self.a1, self.t1, self.t2, dt, "abs_dt")
+        assert 0 == x.size == y.size
```

Comment:
So I know we said we'd hold off on turning this into a loop since 2 things were changing (1st and 4th arguments), but looking at this a bit more closely I found that for the fourth argument, we have `t2` and `t1 * 10`. Since `t2 = a1 * 10` and `t1 = a1`, these are equivalent. So we should be able to fix that as `t2` and loop only changing the first argument

Reply:
Updated